src/debuginfo.cpp

```diff
@@ -861,10 +880,22 @@ void jl_getFunctionInfo(char **name, size_t *line, char **filename,
         if (*line == (size_t) -1) {
             *line = prev.Loc.getLine();
         }
+
+        DILexicalBlockFile locscope = DILexicalBlockFile(prev.Loc.getScope((*it).second.func->getContext()));
+        jl_copy_str(filename, (char*)(uintptr_t)locscope.getFilename().data());
```

Comment:
Is this valid? IIUC, `llvm::StringRef::str()` returns a [new `std::string` object](http://llvm.org/docs/doxygen/html/classllvm_1_1StringRef.html#ab02cb6f4884b5c788efb97b6741dccaf) which should get out of scope and be destructed after `c_str()` returns.

I've never dig into the livetime of c++ temporaries so I might be wrong here.